lib/internal/modules/esm/hooks.js

```diff
@@ -749,7 +780,25 @@ function nextHookFactory(chain, meta, { validateArgs, validateOutput }) {
         ObjectAssign(meta.context, context);
       }
 
-      const output = await hook(arg0, meta.context, nextNextHook);
+      const esmLoader = require('internal/process/esm_loader').esmLoader;
```

Comment:
@GeoffreyBooth If the PR was to be split then this block would probably be the main chunk to split out. I think the major issue of splitting this block out is that it is now re-used in _BOTH_ static and dynamic loader imports and splitting this might mean that static import chaining will break 😢

Reply:
Okay, I don't think it's so important that it be split. We just try to make each PR as discrete as possible.

Reply:
Okay great I'll mark this as resolved!